### PR TITLE
feat(essentials): add `preferReuse` setting

### PR DESCRIPTION
**Pull request by marionebl**
**What's the problem this PR addresses?**

 I want to keep dependencies across all workspaces in sync as much as possible; working with a larger number of collaborators often means `preferInteractive` is not enough to reuse existing versions. `--cached` fits the bill but there is no option to enable it by default.

E.g. for a scenario where `react@16` is present somewhere in the project I want to install `react@16` over `react@17` given the command

```
yarn add react
```

...

**How did you fix it?**

* Added a `preferReuse` options

...

**Checklist**
- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).
- [x] I have set the packages that need to be released for my changes to be effective.
- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

File: packages/plugin-essentials/sources/commands/add.ts [hunk 1]
```diff
@@ -133,19 +133,23 @@ export default class AddCommand extends BaseCommand {
...
-        suggestUtils.Strategy.CACHE,
-      ] : [],
+
+      this.cached ?
+        suggestUtils.Strategy.CACHE
+        : undefined,
+
       suggestUtils.Strategy.LATEST,
-    ];
+    ].filter(isStrategy);
```
Comment (merceyz) [inline packages/plugin-essentials/sources/commands/add.ts:152]:
```suggestion
    ].filter((strategy): strategy is suggestUtils.Strategy => typeof strategy !== `undefined`);
```
I would prefer to have this inlined here

File: packages/plugin-essentials/sources/commands/add.ts [hunk 2]
```diff
@@ -323,6 +327,10 @@ export default class AddCommand extends BaseCommand {
   }
 }
 
+function isStrategy(strategy?: suggestUtils.Strategy): strategy is suggestUtils.Strategy {
+  return typeof strategy !== `undefined`;
+}
+
```
Comment (merceyz) [inline packages/plugin-essentials/sources/commands/add.ts:333]:
```suggestion
```

**Review by merceyz — CHANGES_REQUESTED** [review]
Thanks for the PR, I really appreciate that you included a test and updated the JSON schema.
I have a few comments but otherwise this looks good.

File: packages/plugin-essentials/sources/index.ts [hunk 3]
```diff
@@ -125,6 +126,12 @@ const plugin: Plugin = {
       values: [`^`, `~`, ``],
       default: suggestUtils.Modifier.CARET,
     },
+
+    preferReuse: {
+      description: `If true, the CLI will use the hightest version of existing, latest for new dependencies`,
```
Comment (merceyz) [inline packages/plugin-essentials/sources/index.ts:131]:
```suggestion
      description: `If true, `yarn add` will attempt to reuse the most common dependency range in other workspaces.`,
```

**Review by merceyz — CHANGES_REQUESTED** [review]

**Comment by arcanis** [comment]
LGTM! Thanks @marionebl!

**Pull request merged**